### PR TITLE
Avoid the use of os.Exit in tests

### PR DESCRIPTION
**Pull request by sevein**
I'm suggesting to pass *testing.T to pubKeyHandler using a closure so we can interrupt the test without using os.Exit.

Related: https://github.com/uber-go/guide/blob/master/style.md#exit-in-main.

### CODE REVIEW

**Comment by djjuhasz** [comment]
> I'm suggesting to pass *testing.T to pubKeyHandler using a closure so we can interrupt the test without using os.Exit.
> 
> Related: https://github.com/uber-go/guide/blob/master/style.md#exit-in-main.

How can I pass *testing.T to pubKeyHandler? it has to match the PublicKeyHandler signature: https://pkg.go.dev/github.com/gliderlabs/ssh#PublicKeyHandler

Are you suggesting I use a closure or something?

**Comment by sevein** [comment]
Yeah exatly, a closure helps us to meet the signature while we run another function:

```go
		PublicKeyHandler: func(ctx ssh.Context, key ssh.PublicKey) bool {
			return pubKeyHandler(t, ctx, key)
		}
```

File: internal/sftp/goclient_test.go [hunk 1]
```diff
@@ -24,27 +24,28 @@ import (
...
 	}
+
 	defer file.Close()
 
 	scanner := bufio.NewScanner(file)
 	for scanner.Scan() {
 		allowed, _, _, _, err := ssh.ParseAuthorizedKey([]byte(scanner.Text()))
 		if err != nil {
-			log.Fatalf("SFTP server: couldn't parse authorized keys: %s", err)
+			t.Fatalf("SFTP server: couldn't open authorized_keys file: %s", err)
```
Comment (djjuhasz) [inline internal/sftp/goclient_test.go]: I think this is a copy-paste error. Should be:

```
t.Fatalf("SFTP server: couldn't parse authorized keys: %s", err)
```

**Review by djjuhasz — CHANGES_REQUESTED** [review]
Looks good overall @sevein , I think there is just one copy-paste mistake.

**Pull request merged**